Unable to consume latest dotnet/runtime

<!--This is just a template - feel free to delete any and all of it and replace as appropriate.-->

### Description

Windows Forms 6.0 build is failing unable to take latest updates from dotnet/runtime
	
* PR: https://github.com/dotnet/winforms/pull/3804
* Build: https://dev.azure.com/dnceng/public/_build/results?buildId=805275&view=logs&jobId=1f132584-ab08-5fbf-e5d0-5685de454342&j=1f132584-ab08-5fbf-e5d0-5685de454342&t=1ce98a45-1f3f-5914-0114-7502bfbbe8a4
	 

```
##[error]local_define(1,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) syntax error at token '┐' in: ╗∩System.Runtime 

error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command "F:\workspace\_work\1\s\.packages\\runtime.win-x64.microsoft.netcore.ilasm\6.0.0-alpha.1.20454.24\runtimes\win-x64\native/ilasm -QUIET -NOLOGO -RESOURCES=F:\workspace\_work\1\s\artifacts\obj\Accessibility\Debug\netcoreapp5.0\Accessibility.ref.res.obj -DLL -OUTPUT="F:\workspace\_work\1\s\artifacts\obj\Accessibility\Debug\netcoreapp5.0\Accessibility.dll" -KEY="F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20452.19\tools\snk/35MSSharedLib1024.snk" Accessibility.il" exited with code 1.
```


### Regression?

Possibly?

--- a/src/coreclr/src/ilasm/grammar_after.cpp
+++ b/src/coreclr/src/ilasm/grammar_after.cpp
@@ -924,10 +924,6 @@ Its_An_Id:
                                     tok = parse_literal(curSym, curPos, FALSE);
                                     if(tok == QSTRING)
                                     {
-                                        // insert UTF-8 BOM prefix
-                                        yylval.binstr->insertInt8(0xEF);
-                                        yylval.binstr->insertInt8(0xBB);
-                                        yylval.binstr->insertInt8(0xBF);
                                         yylval.binstr->appendInt8(' ');
                                         DefineVar(newstr, yylval.binstr);
                                     }

--- a/src/tests/ilasm/System/Runtime/CompilerServices/MiAggressiveOptimization.il
+++ b/src/tests/ilasm/System/Runtime/CompilerServices/MiAggressiveOptimization.il
@@ -1,13 +1,15 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-.assembly extern System.Runtime {}
+#define CORE_ASSEMBLY "System.Runtime"
+
+.assembly extern CORE_ASSEMBLY {}
 .assembly MiAggressiveOptimization {}
 .module MiAggressiveOptimization.dll
 // MVID: {B5A4E68E-19B7-4FA3-9BE2-2A0B7265233B}
 
 .class public abstract auto ansi sealed beforefieldinit Program
-       extends [System.Runtime]System.Object
+       extends [CORE_ASSEMBLY]System.Object
 {
   .method public hidebysig static void  Main() cil managed aggressiveoptimization
   {

